pkg/util/jsonpath/jsonpath.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jsonpath
+
+import (
+	"errors"
+	"github.com/GoogleCloudPlatform/kubernetes/third_party/golang/parse"
+	"github.com/golang/glog"
+	"io"
+	"reflect"
+	"strconv"
+)
+
+type Jsonpath struct {
+	name string
+	tree *parse.Tree
+}
+
+func New(n string) *Jsonpath {
```

Comment:
Given that all this is construct the object, why bother with a New function, just make Name a public field in the struct.